### ISSUE TITLE
Question on Asymmetric Query-Document Setup (Long Queries, Short Documents)

### ISSUE THREAD

**Issue by yaswanth19** [UKPLab/sentence-transformers#3311]
Hi,

I’m working on a problem where the query inputs are long (e.g., detailed product reviews or user histories) and the documents are very short (e.g., product titles or short descriptions). This setup is inherently asymmetric compared to usual retrieval problem statements, and while it could be modeled as a multi-label classification problem, we are exploring information retrieval (IR)-based approaches due to the extremely large label space (approx. 10k classes).

I went through the sentence-transformers documentation but couldn't find any example of this sort hence requesting the community to give some reference 🙏 .

**Comment by tomaarsen** [comment]
Hello!

If your goal is that users can pass in detailed product reviews or user histories, and that they'll be shown short documents (product titles, short descriptions) much like a search system, then a IR solutions should definitely be considered. 

You'll definitely be able to use the [standard training loop](https://sbert.net/docs/sentence_transformer/training_overview.html), but with your question-answer pairs (but a long question and short answer). I would start with a model like [ModernBERT-base](https://huggingface.co/answerdotai/ModernBERT-base) that can handle long texts pretty easily. Or an already finetuned one like [Alibaba-NLP/gte-modernbert-base](https://huggingface.co/Alibaba-NLP/gte-modernbert-base).

The [primary training script in the docs](https://sbert.net/docs/sentence_transformer/training_overview.html#trainer) should then work out of the box.

If your data is differently structured (i.e. you don't have query-answer pairs), then you can also have a look at the [Loss Overview](https://sbert.net/docs/sentence_transformer/loss_overview.html) docs to check if there's another loss function that does match your data format. However, query-answer pairs are often most convenient to make.

- Tom Aarsen

**Comment by yaswanth19** [comment]
@tomaarsen Thank you!!
